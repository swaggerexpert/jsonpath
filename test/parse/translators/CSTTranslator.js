import { assert } from 'chai';
import { jestExpect as expect } from 'mocha-expect-snapshot';

import { parse, CSTTranslator } from '../../../src/index.js';

describe('parse', function () {
  context('translators', function () {
    context('CSTTranslator', function () {
      specify('should translate a JSONPath to a CST', function () {
        const parseResult = parse('$', { translator: new CSTTranslator() });

        assert.isTrue(parseResult.result.success);
        expect(parseResult.tree).toMatchSnapshot();
      });
    });
  });
});
