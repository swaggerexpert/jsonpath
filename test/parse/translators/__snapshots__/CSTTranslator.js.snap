// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse translators CSTTranslator should translate a JSONPath to a CST 1`] = `
{
  "root": {
    "type": "jsonpath-query",
    "text": "$",
    "start": 0,
    "length": 1,
    "children": [
      {
        "type": "root-identifier",
        "text": "$",
        "start": 0,
        "length": 1,
        "children": [],
      },
    ],
  },
}
`;
